upstream-tag-sync/src/index.ts

```diff
@@ -77,141 +224,71 @@ async function run(): Promise<void> {
-        branchName,
-        '--force-with-lease'
-    ]);
 
     // Get repository default branch
-    const { data: repo } = await octokit.rest.repos.get({
-        owner: targetOwner,
-        repo: targetRepoName
-    });
-    const defaultBranch = repo.default_branch;
-
-    // Common parts of the PR message
-    const commonHeader = `This PR ${hasConflicts ? 'attempts to' : ''} sync your fork with the upstream repository's tag ${latestTag}.`;
-    const commonChanges = `## Changes included:
-- ${hasConflicts ? 'Attempted merge' : 'Successfully merged'} with tag ${latestTag}
-- Updates from: https://github.com/${upstreamRepo}`;
-    const commonFooter = `You can safely delete the \`${branchName}\` branch afterward.`;
-
-    // Create PR with appropriate message
-    const conflictInstructions = `## ⚠️ Merge Conflicts Detected
-This PR contains merge conflicts that need to be resolved manually. Please:
-1. Checkout this branch locally
-2. Resolve the conflicts
-3. Push the resolved changes back to this branch
-
-### Next Steps:
-1. Resolve conflicts between your customizations and upstream changes
-2. Once conflicts are resolved:
-   - If you want to sync to this tag: merge the PR
-   - If you don't want to sync: close the PR
-3. ${commonFooter}`;
-
-    const normalInstructions = `Please review the changes and:
-- If you want to sync to this tag: merge the PR
-- If you don't want to sync: close the PR
-
-${commonFooter}`;
-
-    const prBody = `${commonHeader}
-
-${hasConflicts ? conflictInstructions : normalInstructions}
-
-${commonChanges}`;
-
-    const { data: pr } = await octokit.rest.pulls.create({
-        owner: targetOwner,
-        repo: targetRepoName,
-        title: hasConflicts
-            ? `[Conflicts] Sync with upstream tag ${latestTag}`
-            : `Sync with upstream tag ${latestTag}`,
-        head: branchName,
-        base: defaultBranch,
-        body: prBody
-    });
-
-    // Add appropriate labels
-    const labels = [labelName];
-    if (hasConflicts) {
-        labels.push('merge-conflicts');
-    }
+    const defaultBranch = await getDefaultBranch(octokit, targetOwner, targetRepoName);
 
-    await octokit.rest.issues.addLabels({
-        owner: targetOwner,
-        repo: targetRepoName,
-        issue_number: pr.number,  // Note that PRs are also issues on GitHub
-        labels: labels
-    });
+    // Set up git configuration
+    await exec('git', ['config', '--global', 'user.name', 'github-actions[bot]']);
+    await exec('git', ['config', '--global', 'user.email', 'github-actions[bot]@users.noreply.github.com']);
+
+    // Add upstream remote and fetch the specific tag
+    await exec('git', ['remote', 'add', 'upstream', `https://github.com/${upstreamRepo}.git`]);
+    await exec('git', ['fetch', 'upstream', `refs/tags/${latestTag}:refs/tags/${latestTag}`, '--no-tags']);
+
+    // Create and push the branch
+    await exec('git', ['checkout', '-b', branchName, latestTag]);
```

Comment:
Note to self: this assumes that the target `branchName` doesn't exist (which it shouldn't, but a side effect is that we won't update the branch if the corresponding upstream tag was updated)

Reply:
Yes that is true but I would say that updating an existing tag should be considered really bad practice. Anyway I've added a comment.